hal-x86_64/src/cpu.rs

```diff
@@ -0,0 +1,59 @@
+use core::mem;
+
+#[repr(transparent)]
+pub struct Port {
+    num: u16,
+}
+
+impl Port {
+    pub const fn at(address: u16) -> Self {
+        Port { num: address }
+    }
+
+    pub unsafe fn readb(&self) -> u8 {
+        let result: u8;
+        asm!("inb %dx, %al" : "={al}"(result) : "{dx}"(self.num) :: "volatile");
+        result
+    }
```

Comment:
clippy wants a safety doc here but all i can say is `# Safety: if you use this wrong the computer might blow up or melt`

do you think `Safety: safety of these functions is entirely dependent on correct interaction with the device these port operations communicate with` is helpful?